cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -70,9 +73,12 @@ jsonFromMetadataValue md =
         Map _ -> Nothing
         List _ -> Nothing
         I _ -> Nothing
-        B bs -> Just ("0x" <> Text.decodeUtf8 (Base16.encode bs), jsonFromMetadataValue b)
+        B _ -> Nothing
```

Comment:
Ah, this is tricky because the spec actually denotes that all of these metadata value types would be valid keys. 

That being said, it's probably the right choice to disallow these metadata value types for the JSON representation of tx metadata since it becomes a bit complicated. However, I'm not sure if it's the right decision to silently ignore them as we're doing here.

I think we should address this and also better document it in a follow up PR.

Reply:
Right now, we don't seem to use `jsonFromMetadata` outside of the tests but, if we did, this could be problematic as data could be silently lost when serializing tx metadata as JSON.

Reply:
Hmm, maybe I need better comments here.

Reply:
Better comments added.

Reply:
Oops, disregard my comments. I misunderstood what was going on here.

Regardless, better docs are good :smiley: